src/ASM/ASMs2DLag.C

```diff
@@ -789,6 +782,47 @@ bool ASMs2DLag::evalSolution (Matrix& sField, const IntegrandBase& integrand,
 }
 
 
+bool ASMs2DLag::evalSolution (Matrix& sField, const IntegrandBase& integrand,
+                              const RealArray* gpar, bool) const
+{
+  sField.resize(0,0);
+
+  size_t nPoints = gpar[0].size();
+
+  FiniteElement fe(p1*p2);
+  Vector        solPt;
+  Vectors       globSolPt(nPoints);
+  Matrix        dNdu, Xnod, Jac;
+
+  // Evaluate the secondary solution field at each point
+  for (size_t i = 0; i < gpar[0].size(); ++i) {
+    const int iel = this->findElement(gpar[0][i], gpar[1][i], &fe.xi, &fe.eta);
+
+    const IntVec& mnpc = MNPC[iel-1];
```

Comment:
This temporary is only used once (in L814). Then it is better to just write `MNPC[iel-1]` there

Reply:
sure.